src/test/regress/expected/create_ref_dist_from_citus_local.out

```diff
@@ -0,0 +1,337 @@
+ reference_table_2        | n          | t
+(9 rows)
+
+  SELECT COUNT(*)=12 FROM pg_constraint
+  WHERE connamespace = (SELECT oid FROM pg_namespace WHERE nspname='create_ref_dist_from_citus_local') AND
+        conname ~ '^fkey\_\d+$';
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+BEGIN;
+  -- define a self reference and a foreign key to reference table
+  ALTER TABLE citus_local_table_5 ADD CONSTRAINT fkey_12 FOREIGN KEY (col_1) REFERENCES citus_local_table_5(col_1);
+  ALTER TABLE citus_local_table_5 ADD CONSTRAINT fkey_13 FOREIGN KEY (col_1) REFERENCES reference_table_1(col_1);
+  SELECT create_distributed_table('citus_local_table_5', 'col_1');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  SELECT logicalrelid::text AS tablename, partmethod, repmodel FROM pg_dist_partition
+  WHERE logicalrelid::text IN (SELECT tablename FROM pg_tables WHERE schemaname='create_ref_dist_from_citus_local')
+  ORDER BY tablename;
+        tablename         | partmethod | repmodel
+---------------------------------------------------------------------
+ citus_local_table_1      | n          | c
+ citus_local_table_2      | n          | c
+ citus_local_table_3      | n          | c
+ citus_local_table_4      | n          | c
+ citus_local_table_5      | h          | c
+ distributed_table_1      | h          | c
+ partitioned_dist_table_1 | h          | c
+ reference_table_1        | n          | t
+ reference_table_2        | n          | t
+(9 rows)
+
+  SELECT COUNT(*)=13 FROM pg_constraint
+  WHERE connamespace = (SELECT oid FROM pg_namespace WHERE nspname='create_ref_dist_from_citus_local') AND
+        conname ~ '^fkey\_\d+$';
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+CREATE TABLE citus_local_table_6 (col_1 INT UNIQUE);
+SELECT create_citus_local_table('citus_local_table_6');
+ create_citus_local_table
+---------------------------------------------------------------------
+
+(1 row)
+
+BEGIN;
+  ALTER TABLE citus_local_table_5 ADD CONSTRAINT fkey_12 FOREIGN KEY (col_1) REFERENCES citus_local_table_6(col_1);
+  -- errors out as foreign keys from distributed tables to citus
+  -- local tables are not supported
+  SELECT create_distributed_table('citus_local_table_5', 'col_1');
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+ROLLBACK;
+BEGIN;
+  -- errors out as foreign keys from citus local tables to distributed
+  -- tables are not supported
+  ALTER TABLE citus_local_table_5 ADD CONSTRAINT fkey_12 FOREIGN KEY (col_1) REFERENCES citus_local_table_6(col_1);
+  SELECT create_distributed_table('citus_local_table_6', 'col_1');
+ERROR:  cannot create foreign key constraint since foreign keys from reference tables to distributed tables are not supported
+ROLLBACK;
```

Comment:
error message is wrong as lhs table is actually a citus local table, not a reference table, I can open a seperate pr for that